src/components/NotificationProvider/NotificationProvider.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  NotificationType,
+  NotificationHelper,
+  QueuedNotification,
+  NotifyProviderProps,
+} from "./types";
+import { useLocation } from "react-router-dom";
+import isEqual from "lodash/isEqual";
+import Notification from "../Notification";
+import { info, failure, success, queue } from "./messageBuilder";
+
+const NotifyContext = createContext<NotificationHelper>({
+  notification: null,
+  clear: () => undefined,
+  failure: () => undefined,
+  success: () => undefined,
+  info: () => undefined,
+  queue: () => undefined,
+});
+
+export const NotificationProvider: FC<NotifyProviderProps> = ({ children }) => {
+  const [notification, setNotification] = useState<NotificationType | null>(
+    null
+  );
+  const { state, pathname } = useLocation() as QueuedNotification;
+
+  const clear = () => notification !== null && setNotification(null);
+
+  const setDeduplicated = (value: NotificationType) => {
+    if (!isEqual(value, notification)) {
+      setNotification(value);
+    }
+    return value;
+  };
+
+  useEffect(() => {
+    if (state?.queuedNotification) {
+      setDeduplicated(state.queuedNotification);
+      window.history.replaceState({}, "");
+    } else {
+      clear();
+    }
+  }, [state, pathname]);
+
+  const helper: NotificationHelper = {
+    notification,
+    clear,
+    queue,
+    failure: (title, error, message, actions) =>
+      setDeduplicated(failure(title, error, message, actions)),
+    info: (message, title) => setDeduplicated(info(message, title)),
+    success: (message) => setDeduplicated(success(message)),
+  };
+
+  return (
+    <NotifyContext.Provider value={helper}>{children}</NotifyContext.Provider>
+  );
+};
+
+export function useNotify() {
+  return useContext(NotifyContext);
+}
+
+export const NotificationConsumer: FC = () => {
+  const notify = useNotify();
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (ref.current?.hasAttribute("scrollIntoView")) {
+      ref.current.scrollIntoView({
+        behavior: "smooth",
+        block: "center",
+        inline: "start",
+      });
+    }
+    window.dispatchEvent(new Event("resize"));
```

Comment:
I wonder if this should dispatch a custom event instead? So that it can be intercepted and handled separately than the standard `resize` event. Maybe that way it would be less "hacky"? What do you think?

Reply:
I think we shouldn't throw this event at all - we can bind the `notify.message` as a dependency to a `useEffect` if we want to run something in a caller on changing notifications.

Reply:
Sure, even better.